Sources/Purchasing/Configuration.swift

```diff
@@ -184,3 +186,15 @@ import Foundation
     }
 
 }
+
+private extension Configuration {
+
+    static func validate(apiKey: String) {
+        if !apiKey.hasPrefix(Self.appleKeyPrefix) {
+            Logger.warn(Strings.configure.invalidAPIKey)
```

Comment:
will this issue a warning when using the legacy api key? that one should still work forever

Reply:
I thought this could encourage people to use the new ones, but if we don't care we can check if it's a different one (if it starts with `anythingElse_`

Reply:
Maybe we could have a separate message for "this is definitely the wrong API key, copy/pasted from the wrong platform" vs "hmmm this might be a legacy key, you should get that updated when you get a chance"

Reply:
That makes sense, maybe make the latter only `debug`?